### DISCUSSION TITLE
unable to render superscript in column name in tibble in Quarto

### DISCUSSION THREAD

**Discussion by lizaodawn** [quarto-dev/quarto-cli#5977] [Q&A] [answered]
### Description

Hi, when I am rendering the html output with Quarto, I found the superscript in a column name in a tibble displayed as a plain string. However, when running in Rscript the same code shows the superscript correctly. 

I think it is due to some setting I missed for telling the r chunk to parse as r expression, but I always get the plain string whatever methods I tried. 

The R script code is: 
top_scores_colloc %>% 
  as_tibble() %>%
  select(type, a, PMI, G_signed) %>% 
  arrange(desc(G_signed)) %>%        
  head(30) %>%                     
  kbl(col.names = c("Type", "Frequency", "PMI", r"(Signed $G^2$)")) %>% 
  kable_minimal() %>% 
  scroll_box(height = "400px") %>% print()

And the code in r chunk in Quarto is:
```{r}
#| warning: false
#| echo: false
#| label: tbl-kwtable_PMI
#| tbl-cap-location: top
#| tbl-cap: Top 30 Keywords on Indian Related Text in Natural History (sorted by PMI)

top_scores_kw %>% 
  as_tibble() %>%
  select(type, a, PMI, G_signed) %>% 
  arrange(desc(PMI)) %>%             
  head(30) %>%                       
  kbl(col.names = c("Type", "Frequency", "PMI", r"(Signed $G^2$)")) %>% 
  kable_minimal() %>% 
  scroll_box(height = "400px")
```

The Quarto version I am using is 1.3.361, and I already set the engine: knitr in the YAML front matter.

Many thanks in advance!!

**Comment by mcanouil** [comment]
Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by lizaodawn** [reply]
Yes sure, I made a test table for example as follows. 
And the Git repository of the original project is: https://github.com/lizaodawn/zhuang/tree/papertry


````qmd
---
engine: knitr
title: "test_table"
format: html
---

## Running Code

```{r}
#| warning: false
#| code-fold: true
library(tidyverse)
library(kableExtra)

test_table <- data.frame(
  Column1 = rnorm(1),
  Column2 = rnorm(1),
  Column3 = rnorm(1),
  Column4 = rnorm(1)
)

test_table %>% 
  as_tibble() %>%                    
  kbl(col.names = c("Type", "Frequency", "PMI", r"(Signed $G^2$)")) %>% 
  kable_minimal() %>% 
  scroll_box(height = "400px")
```

The end.
````

When rendering the test code in Quarto, the output look like this:
![image](https://github.com/quarto-dev/quarto-cli/assets/114702796/a7fc90d1-62ad-4d4e-88ed-7c89e2f8d70b)


While running in R script preview, the superscript shows correctly:
![image](https://github.com/quarto-dev/quarto-cli/assets/114702796/86f92760-a9a8-4694-9b87-761c730ca608)

Many thanks!

**Comment by mcanouil** [comment]
See rstudio/rmarkdown#2279.

**Comment by cl-roberts** [comment, marked as answer]
Hi there,

This is more of a knitr question than Quarto. Since you are rendering in html output, knitr is by default trying to print an html-format table. That is, knitr is expecting html tags for formatting and instead you are feeding latex code. So, in the language of html, changing the `kbl()` function call to the following should have your desired effect:

```
kbl(col.names = c("Type", "Frequency", "PMI", "(Signed G<sup>2</sup>)"), escape = FALSE)
```

Don't forget to include `escape = FALSE`. Full `.qmd` example:

````qmd
---
engine: knitr
title: "test_table"
format: html
---

## Running Code

```{r}
#| warning: false
#| code-fold: true
library(tidyverse)
library(kableExtra)

test_table <- data.frame(
  Column1 = rnorm(1),
  Column2 = rnorm(1),
  Column3 = rnorm(1),
  Column4 = rnorm(1)
)

test_table %>% 
  as_tibble() %>%                    
  kbl(col.names = c("Type", "Frequency", "PMI", "(Signed G<sup>2</sup>)"), escape = FALSE) %>% 
  kable_minimal() %>% 
  scroll_box(height = "400px")
```

The end.
````

Alternatively you can simply change the table format in `kbl()`, for example by using the option `format = "pipe"`. This will make `kable` print a table in the style of Pandoc pipe tables, and thus will render latex code.

**Reply by mcanouil** [reply]
The thing is MathJax and other equation libraries support dollar math equation, but Kable don't.
It means you can't have both HTML and LaTeX table with the same command, while other libraries support this.

**Reply by lizaodawn** [reply]
Thank you very much! Either solution you proposed worked well in my case, it helps a lot! I chose to add format = "pipe" for my report in the end.

**Reply by mcanouil** [reply]
As a reminder, the issue is opened for about a year: <https://github.com/rstudio/rmarkdown/issues/2279>, in case you want to watch it for possible update.

**Reply by lizaodawn** [reply]
Sure, thank you very much!
